packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +608,113 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    this.subscribe(this.#onStateChange);
+
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        const networkClients = this.getNetworkClientRegistry();
+        patches.forEach(({ op, path }) => {
+          if (op === 'remove' && path[0] === 'networkConfigurations') {
+            const networkClientId = path[1] as NetworkClientId;
+            delete networkClients[networkClientId];
+          }
+        });
+
+        this.#refreshTrackingMap(networkClients);
+        this.#refreshEtherscanRemoteTransactionSources(networkClients);
+      },
+    );
+
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      // TODO(JL): Network state changes also trigger PendingTransactionTracker's onStateChange.
+      // Verify if this is still necessary when the feature branch is being reviewed
```

Comment:
this comment can be removed

Reply:
i'm still not sure why a network state change needs to be tied to the PendingTransactionTracker which uses it for start/stop based on if there are any pending tx in state. Still seems odd

Reply:
I don't believe this is necessary, but I don't want to remove it just in case lol

Reply:
https://github.com/MetaMask/core/pull/3865